### DISCUSSION TITLE
Planning

### DISCUSSION THREAD

**Discussion by DizkoDan** [psy0rz/zfs_autobackup#139] [Q&A] [answered]
Hi,

I'm in the planning phase of a new backup strategy for a newly setup ZFS based server (Proxmox). Being new to both, I'm probably just overcomplicating things, which I have a tendency to do. I have a mirrored pair of nvme's running the server. I also have a stand alone sata drive that I intend to do local backups to. I have a new server setup to act as a remote backup server for this, and other servers that I manage. My idea is: 

- Setup the local backup drive as a single disk zfs pool. 
- Snapshot to the local backup pool,
- Send backup from there to the remote backup server.

I obviously want deduplication/incremental backups here. It seems like zfs autobackup would handle this for me, but I'm unsure of the additional layer of the local backup pool/drive. Is there a way to send to multiple targets? Or is there another way that I should be looking at this?

Thanks,
-Dan

**Comment by Scrin** [comment, marked as answer]
This sounds somewhat similar to my setup so perhaps that can be of some inspiration.

In my setup, I have similarly "server(s) to backup", "local backups" and "remote backups". Since the "local backups" are practically "free", I do those frequently (to minimize losses in case of a catastrophic failure on a server that takes out a number of disks at once), and then I do these "remote (offsite) backups" based on these "local backups", but less frequently.

This backup setup consists of two "individual" parts;
1) a "backupper" machine will run backups daily from the server(s) to the "local backups" zfs pool (I'm using the ["pull+push"](https://github.com/psy0rz/zfs_autobackup/wiki#pullpush-zero-trust) strategy, but that's not relevant)
2) another setup is backing up the "local backups" to the "remote backups" location weekly/monthly (less critical data has its "offsite backups" updated less frequently)

These two "parts" are not even aware of each other, simplifying the setup for me. Due to the fact that ZFS snapshots are incremental and the only "space" they take is (practically) just the delta; the amount of data changed between two snapshots, this setup works very well. Having "unnecessary" snapshots "interleaved" in between (on the local backups pool in particular) is not an issue at all, as makes practically zero difference in space usage.

Another benefit in my setup is that the "remote (offsite) backups" are managed from only one place rather than going from each server to the remote on their own.

**Reply by DizkoDan** [reply]
Thanks for the reply. Are you saying that you run multiple zfs-autobackup "jobs"? One to do the local snap/backup, then another that does a snap/backup of the local backup to the remote?

**Reply by Scrin** [reply]
For the purpose of this discussion, yes (in reality my setup is much more complicated).

Just make sure that these two "jobs" use different names (in the property you use to [select which datasets to backup](https://github.com/psy0rz/zfs_autobackup/wiki#select-filesystems-to-backup)), for example use "autobackup:local" on the "servers" for the first job, and "autobackup:offsite" on the "local backups pool" for the 2nd job. This is useful to set on the "parent" dataset so it's automatically inherited to all child datasets (the ones that came from "the servers").

**Reply by DizkoDan** [reply]
Thanks, that's helpful. However, I'm looking at the command syntax, and I don't see how I would do a local snapshot to the backup drive. Seems like it expects an SSH source or target. Am I missing something?

**Reply by psy0rz** [reply]
you can just omit the ssh stuff so it operates locally :)

**Reply by DizkoDan** [reply]
Thanks a lot! I got this working with minimal effort. I also added a section in the [Getting Started](https://github.com/psy0rz/zfs_autobackup/wiki#local-usage) wiki for local usage since there doesn't seem to be any documentation on it. I meant to PR it, but I edited it straight, thinking it would ask 🤷‍♂️

**Reply by psy0rz** [reply]
lol thanks. ive changed the setting to colaborators only. its a bit scary everyone can change it, since i also dont get a notification when someone does.
